web/App.js

```diff
@@ -0,0 +1,235 @@
+const localIP = window.location.hostname;
+
+const socket = new WebSocket('ws://' + localIP + ':7080');
+
+socket.binaryType = 'arraybuffer';
+
+let socketIsReady = false;
+let audioAllowed = false;
+
+socket.onopen = function(e) {
+    socketIsReady = true;
+};
+
+const canvas = document.getElementById("game");
+const ctx = canvas.getContext('2d');
+
+let windowWidth = window.innerWidth;
+let windowHeight = window.innerHeight;
+
+const originalWidth = 320;
+const originalHeight = 180;
+const scaleFactor = Math.floor(windowWidth / originalWidth);
+const horizontalScale = originalWidth * scaleFactor;
+const verticalScale = originalHeight * scaleFactor;
+
+canvas.height = verticalScale;
+canvas.width = horizontalScale;
+
+let mouseDown = false;
+const keysDown = {};
+
+let audioCtx, source;
+
+const gameAssets = {};
+
+const imageCache = {};
+
+socket.onmessage = function(msg) {
+    const buf = new Uint8ClampedArray(msg.data);
+    let color, startX, startY, width, height;
+    let i = 0;
+    while (i < buf.length) {
+        const frameType = buf[i];
+
+        if (frameType === 1) {
+            const assetType = buf[i + 1];
```

Comment:
This is a weird combination of the inverse of the "squish" function in `Squisher.js` & asset decoding/storing. 

I would like to have one common module that can be shared by both the web app & the server that will squish/unsquish classes of thing. Kind of like a common "game node serializer". That's the intent of Squisher, really.